src/app/components/form/DualCurrencyField/index.tsx

```diff
@@ -53,19 +216,20 @@ export default function DualCurrencyField({
         "block w-full placeholder-gray-500 dark:placeholder-gray-600 dark:text-white ",
         "px-0 border-0 focus:ring-0 bg-transparent"
       )}
-      placeholder={placeholder}
+      placeholder={inputPlaceHolder}
       required={required}
       pattern={pattern}
       title={title}
-      onChange={onChange}
+      onChange={onChangeWrapper}
       onFocus={onFocus}
       onBlur={onBlur}
-      value={value}
+      value={inputValue ? inputValue : ""}
       autoFocus={autoFocus}
       autoComplete={autoComplete}
       disabled={disabled}
-      min={min}
-      max={max}
+      min={minValue}
+      max={maxValue}
+      step={useFiatAsMain ? "0.01" : "1"}
```

Comment:
what if we keep step value upto 4 decimal places. so that user can enter values for 10 20 sats in USD as well (it has no major use but just to keep it fully flexible)